blackbird_python/blackbird/__init__.py

```diff
@@ -74,20 +104,85 @@
   If the operation argument you are parsing is an instance of this class,
   that indicates that a register transform is required.
 
-* :func:`~.parse_blackbird`: a utility function that automates
-  the parsing of the Blackbird abstract syntax tree. Use this function, either
-  with the default parser :class:`~.BlackbirdListener` or a custom
-  derived class, to parse arbitrary Blackbird code.
 
 .. note::
 
-	While both the above classes/functions are located in the :mod:`blackbird.listener`,
-	they are both importable from the top-level of the Blackbird Python package:
+  All the above classes/functions are importable from the top-level
+  of the Blackbird Python package:
 
-	.. code-block:: python
+  .. code-block:: python
 
-		from blackbird import BlackbirdListener, parse_blackbird
-"""
+    from blackbird import BlackbirdListener, load, BlackbirdProgram
 
-from .listener import BlackbirdListener, RegRefTransform, parse_blackbird
+Summary
+-------
+
+.. autosummary::
+    load
+    loads
+    dump
+    dumps
+
+Code details
+^^^^^^^^^^^^
+"""
+import antlr4
+
+from .listener import (
+    BlackbirdListener,
+    RegRefTransform,
+    parse
+)
+from .program import BlackbirdProgram
 from ._version import __version__
+
+
+def load(filename):
+    """Deserialize a blackbird program from a file to a
+    :class:`BlackbirdProgram` object.
+
+    Args:
+        filename (str): file location of a valid Blackbird program
+
+    Returns:
+        BlackbirdProgram: returns a :class:`BlackbirdProgram` object
```

Comment:
These types of comments unnecessarily repeat both the name of the returned class and the word "returns", and don't actually say anything not already present in the napoleon syntax :)
```suggestion
        BlackbirdProgram: parsed representation of the program
```

Reply:
(also happens elsewhere in the code)

Reply:
This is true. I had a look at the Python JSON and YAML libraries, and they use docstrings without any `Args:` and `Returns:` fields.